### PR TITLE
Fix invalid collection for Interface Display Template for M2M relationships

### LINKED ISSUE #9419: Invalid collection in Interface Display Template for m2m fields
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

When configuring a m2m field, Display Template field inside Interface tab is not configured with the right collection. It should be configured with the same collection as the Display tab, but it's not.

### To Reproduce

Try to configure a Display Template inside Interface tab on a m2m collection. The tree root is not matching expected data to build the template.

### Errors Shown

_No response_

### What version of Directus are you using?

v9.0.0-dev

### What version of Node.js are you using?

v16.13.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying Directus?

locally (dev)

### PR DESCRIPTION
**Pull request by Toilal**
Fixes #9419

### CODE REVIEW

File: app/src/interfaces/list-m2m/index.ts [hunk 1]
```diff
@@ -14,14 +14,16 @@ export default defineInterface({
 	localTypes: ['m2m'],
 	group: 'relational',
 	options: ({ relations }: ExtensionsOptionsContext) => {
+		const relatedCollection = relations.o2m?.collection ?? relations.m2o?.related_collection;
```
Comment (Toilal) [inline app/src/interfaces/list-m2m/index.ts:17]: I have copied the code from `displays/relative-values/index.ts` that seems to work properly.

Comment (joselcvarela) [reply app/src/interfaces/list-m2m/index.ts:17]:
```suggestion
		const relatedCollection = relations.m2o?.related_collection ?? relations.o2m?.collection;
```

To be compatible with https://github.com/directus/directus/pull/9391/files#diff-dcb0b7b8e24638261fb003d9ae4222a524a0da95c57c9a5997f0a65134dfe202R20

**Review by joselcvarela — APPROVED** [review]

File: app/src/interfaces/list-m2m/index.ts (hunk 1 above)
Comment (rijkvanzanten) [reply app/src/interfaces/list-m2m/index.ts:17]: The related collection for a m2m field should always be the o2m relation collection. We can remove the m2o one here

Comment (joselcvarela) [reply app/src/interfaces/list-m2m/index.ts]: Fixed

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**